third_party/utf8_range/utf8_range.h

```diff
@@ -0,0 +1,9 @@
+
+#if defined(__ARM_NEON) || defined(__SSE4_1__)
+int utf8_range2(const unsigned char *data, int len);
+#else
+int utf8_naive(const unsigned char *data, int len);
+static inline int utf8_range2(const unsigned char *data, int len) {
```

Comment:
`static inline` in a header file feels like an invitation for ODR violations if anyone ever takes the address of it

Reply:
This is a good resource that goes through C's inline rules and possible approaches. upb is using strategy (1): http://www.greenend.org.uk/rjk/tech/inline.html

Reply:
From that link to parts come together for my concern:

- "(C++ is stricter: a function which is inline anywhere must be inline everywhere and must be defined identically in all the translation units that use it.)"
- In the strategy 1 section: "if you take the address of the function in two translation units then the result won't compare equal"

These two facts combine to make `static inline` ODR bait for folks taking the address of them.  I don't have a better answer given C's lack of stripping but that is the crux of my concern.